src/commands/build/variants.ts

```diff
@@ -0,0 +1,71 @@
+import chalk from "chalk";
+import { getAllVariantsInPath } from "../../files/variants/getAllPackageVariants.js";
+
+/**
+ * Main function to retrieve the valid variant names based on the specified variants and available directories.
+ */
+export function getVariantNames({
```

Comment:
I would rename it to `getValidVariantNames` just to make it a little bit clearer